core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,663 @@
+const addStateInput = '.e2e-test-add-state-input';
+const saveOutcomeDestButton = '.e2e-test-save-outcome-dest';
+const closeMetadataModal = '.e2e-test-close-history-metadata-modal';
+const closeStateModal = '.e2e-test-close-history-state-modal';
+const oppiaResponseSelector = '.oppia-click-to-start-editing';
+
+// Preview tab elements.
+const testContinueButton = '.e2e-test-continue-button';
+const testFloatFormInput = '.e2e-test-float-form-input';
+const nextCardButton = '.e2e-test-next-card-button';
+const submitAnswerButton = '.e2e-test-submit-answer-button';
+const explorationRestartButton = '.e2e-preview-restart-button';
+const explorationConversationContent = '.e2e-test-conversation-content';
+
+// Settings tab elements.
+const explorationTitleInput = '.e2e-test-exploration-title-input';
+const explorationCompletedMessage: string = 'div.toast-message';
+
+export class ExplorationEditor extends BaseUser {
+  /**
+   * Function to navigate to the creator dashboard page.
+   */
+  async navigateToCreatorDashboard(): Promise<void> {
+    await this.page.goto(creatorDashboardAdminUrl);
+  }
+
+  /**
+   * Function to create an exploration in the Exploration Editor.
+   * @param {string} content - The content in the Exploration.
+   */
+  async createExploration(content: string): Promise<void> {
+    await this.clickOn(createExplorationButtonSelector);
+    await this.page.waitForSelector(dismissWelcomeModalSelector, {
+      visible: true,
+    });
+    await this.clickOn(dismissWelcomeModalSelector);
+    await this.page.waitForTimeout(300);
+    await this.clickOn(stateEditSelector);
+    await this.page.waitForSelector(explorationContentInput, {visible: true});
+    await this.page.type(explorationContentInput, `${content}`);
+    await this.clickOn(saveContentButton);
+  }
+
+  /**
+   * Function to add an interaction to the exploration.
+   * @param {string} interactionToAdd - The interaction type to add to the Exploration.
+   */
+  async addAnInteractionToTheExploration(
+    interactionToAdd: string
+  ): Promise<void> {
+    await this.page.waitForSelector(addInteractionButton);
+    await this.clickOn(addInteractionButton);
+    await this.page.waitForSelector(endInteractionSelector, {visible: true});
+    await this.clickOn(interactionToAdd);
+    await this.clickOn(saveInteractionButton);
+    if (interactionToAdd !== ' End Exploration ') {
+      await this.clickOn('.oppia-response-header');
+    }
+  }
+
+  /**
+   * Function to save an exploration draft.
+   */
+  async saveExplorationDraft(): Promise<void> {
+    await this.clickOn(saveChangesButton);
+    await this.page.waitForSelector(saveDraftButton, {visible: true});
+    await this.page.waitForTimeout(500);
+    await this.clickOn(saveDraftButton);
+  }
+
+  /**
+   * Function to navigate to the settings tab in the Exploration Editor.
+   */
+  async navigateToSettingsTab(): Promise<void> {
+    await this.page.waitForTimeout(200);
```

Comment:
We won't accept PRs that use waitForTimeout(). Always wait for a specific event to occur, similar to the waitFor in the webdriverio e2e tests.

Reply:
Sure, taken care.